src/Fix64.cs

```diff
@@ -382,66 +395,68 @@ public static Fix64 Exp(Fix64 x)
              * When the sum term drops to zero, we can stop summing.
              */
 
-            // The power-series converges much faster on positive values
-            // and exp(-x) = 1/exp(x).
-            bool neg = (x.RawValue < 0);
-            if (neg) x = -x;
-
-            Fix64 result = x + One;
-            Fix64 term = x;
+            int integerPart = (int)Floor(x);
+            x = FractionalPart(x);
 
-            for (int i = 2; i < 40; i++)
+            Fix64 result = One;
+            Fix64 term = One;
+            int i = 1;
+            while (term.m_rawValue != 0)
             {
-                term = x * term / (Fix64)i;
+                term = x * term * Ln2 / (Fix64)i;
                 result += term;
-
-                if (term.RawValue == 0)
-                    break;
+                i++;
             }
 
+            result = FromRaw(result.m_rawValue << integerPart);
             if (neg) result = One / result;
 
             return result;
         }
 
-        public static Fix64 Ln(Fix64 x)
+        public static Fix64 Log2(Fix64 x)
```

Comment:
I wouldn't expose Log2 as public API. A more useful API (which can be a separate PR) would be Log(a, newBase) like System.Math offers.